tests/codegen/waiter-tests/model/function-type.smithy

```diff
@@ -0,0 +1,169 @@
+$version: "2"
+namespace com.test
+
+use smithy.waiters#waitable
+
+@suppress(["WaitableTraitJmespathProblem"])
+@waitable(
+    TypeFunctionStringEquals: {
+        acceptors: [
+            {
+                state: "success",
+                matcher: {
+                    output: {
+                        path: "type(primitives.string) == types.string",
+                        expected: "true",
+                        comparator: "booleanEquals"
+                    }
+                }
```

Comment:
**Question**: The matcher conditions in this file seem odd to me. Why compare against another structure's members when we have to explicitly set their values? Would `path: "type(primitives.string) == \"string\""` be simpler? Or even something that uses `stringEquals` directly?

```smithy
matcher: {
    output: {
        path: "type(primitives.string)",
        expected: "string",
        comparator: "stringEquals"
    }
}
```

Reply:
My bad, I forgot about `stringEquals` for some reason